sbor/src/print/tx_summary_detector.rs

```diff
@@ -1,3 +1,8 @@
+/// Transaction summary detector is used to determine the type of the transaction intent and collect
+/// information about fees.
+/// Implementation consists of two independent state machines - one for detecting the intent type and
+/// other to collect fee information. Both of them use information about decoded instructions
+/// received from `InstructionExtractor`.
```

Comment:
incorrect doc, either use `//!` for documenting the whole file, or attach the doc to some specific type.